### PR TITLE
Fixing a sporadic timing error prevents a backup from being written correctly to a file

### PR DESCRIPTION
**Pull request by mrRobot62**
On mac I found a sporadic timing error which prevents to write a complete backup into a file. It must be something with the connection between FC and BF configurator, if there is a delay there, the case occurs

Problem was found here
`       GUI.interval_add(readingDumpIntervalName, () => {
            const currentTime = performance.now();
            if (currentTime - lastCliStringReceived > 500) {
                this.cliEngine.unsubscribeOnRowCame();
                GUI.interval_remove(readingDumpIntervalName);
                resolve(diffAll);
            }
`
Unfortunately increasing 500 to a bigger number do not fix this issue.
Sometimes it returns more or less nothing.

Based on a sporadic error my fix is not high spophisticated. In function `TABS.presets.onSaveConfigClick` I check text length, if it is to small, quit function with a warning otherwise write file.

Change `diff all` to `dump all` based on function name which is called.

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Comment by haslinghuis** [comment]
@mrRobot62 please see also: https://github.com/betaflight/betaflight-configurator/pull/2879

**Comment by limonspb** [comment]
@mrRobot62 could you please test the latest master and check if we need this fix from you now. Thanks!

File: src/tabs/presets/presets.js
Comment (limonspb) [inline src/tabs/presets/presets.js:222]: this should stay Diff not dump. The idea is to actually save diff as it's smaller, faster to save/load and more flexible.

**Review by limonspb — CHANGES_REQUESTED** [review]
@mrRobot62 could you please test the latest master and check if we need this fix from you now. Thanks!

**Pull request closed without merging**
